### PR TITLE
quba: init at 1.4.0

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

add [quba](https://github.com/ZUGFeRD/quba-viewer), viewer for electronic invoices

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/qu/quba/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
+
+  meta = with lib; {
+    description = "Viewer for electronic invoices ";
```
Comment (Sigmanificient) [inline pkgs/by-name/qu/quba/package.nix:31]:
```suggestion
    description = "Viewer for electronic invoices";
```

File: pkgs/by-name/qu/quba/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
+
+  meta = with lib; {
```
Comment (Sigmanificient) [inline pkgs/by-name/qu/quba/package.nix:30]: - #208242
```suggestion
  meta = {
```
add `lib.` to `license`, `sourceProvenance` & `maintainers`

File: pkgs/by-name/qu/quba/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    description = "Viewer for electronic invoices ";
+    homepage = "https://github.com/ZUGFeRD/quba-viewer";
+    downloadPage = "https://github.com/ZUGFeRD/quba-viewer/releases";
+    license = licenses.asl20;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    maintainers = with maintainers; [ onny ];
```
Comment (Sigmanificient) [inline pkgs/by-name/qu/quba/package.nix:36]:
```suggestion
    license = lib.licenses.asl20;
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
    maintainers = with lib.maintainers; [ onny ];
```

File: pkgs/by-name/qu/quba/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  appimageContents = appimageTools.extractType1 { inherit name src; };
+in
+appimageTools.wrapType1 {
+  inherit name src;
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
```
Comment (Sigmanificient) [inline pkgs/by-name/qu/quba/package.nix:26]: - #260776
```suggestion
      --replace-fail 'Exec=AppRun' 'Exec=${pname}'
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by onny** [comment]
@Sigmanificient thank you, fixed it!

**Review by Sigmanificient — APPROVED** [review]

**Pull request merged**